From 02918fd53434a23a72fe878a90f4ec48ef0e0416 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Mon, 09 Jan 2012 09:00:40 +0000
Subject: Don't use XFreeXDGA to determine DGA support

If our server supports DGA and we want to build the dummy driver without it,
XFreeXDGA will be defined by the server and will be picked up rather than
our configuration option.  This change forces us to honor our configuration
hoice.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
diff --git a/configure.ac b/configure.ac
index d92e22a..6861485 100644
--- a/configure.ac
+++ b/configure.ac
@@ -61,7 +61,7 @@ XORG_DRIVER_CHECK_EXT(XV, videoproto)
 
 if test "x$DGA" = xyes; then
 	XORG_DRIVER_CHECK_EXT(XFreeXDGA, xf86dgaproto)
-	AC_DEFINE(XFreeXDGA, 1, [Support DGA extension])
+	AC_DEFINE(USE_DGA, 1, [Support DGA extension])
 fi
 AC_SUBST([DGA])
 AM_CONDITIONAL([DGA], [test "x$DGA" = xyes])
diff --git a/src/dummy_driver.c b/src/dummy_driver.c
index 041ca96..6533b27 100644
--- a/src/dummy_driver.c
+++ b/src/dummy_driver.c
@@ -45,7 +45,7 @@
 #include <X11/Xproto.h>
 #include "scrnintstr.h"
 #include "servermd.h"
-#ifdef XFreeXDGA
+#ifdef USE_DGA
 #define _XF86DGA_SERVER_
 #include <X11/extensions/xf86dgaproto.h>
 #endif
@@ -592,7 +592,7 @@ DUMMYScreenInit(int scrnIndex, ScreenPtr pScreen, int argc, char **argv)
 
     xf86SetBlackWhitePixels(pScreen);
 
-#ifdef XFreeXDGA
+#ifdef USE_DGA
     DUMMYDGAInit(pScreen);
 #endif
     
--
cgit v0.9.0.2-2-gbebe
